File: src/main/java/site/gachontable/gachontablebe/domain/pub/domain/Thumbnail.java
package site.gachontable.gachontablebe.domain.pub.domain;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity(name = "thumbnail")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Thumbnail {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer thumbnailId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "pub_id", nullable = false)
    private Pub pub;

    @Column(nullable = false)
    private String url;
}

File: src/main/kotlin/site/gachontable/domain/pub/domain/Thumbnail.kt
package site.gachontable.domain.pub.domain

import jakarta.persistence.*

@Entity(name = "thumbnail")
class Thumbnail(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = 0,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "pub_id", nullable = false)
    val pub: Pub,

    @Column(nullable = false)
    var url: String,
) {
    fun update(url: String) {
        this.url = url
    }

    companion object {
        fun create(url: String, pub: Pub): Thumbnail {
            return Thumbnail(
                url = url,
                pub = pub
            )
        }
    }
}
